cassandra/io/asyncorereactor.py

```diff
@@ -288,9 +291,12 @@ def _cleanup(self):
 
         log.debug("Event loop thread was joined")
 
-        # Ensure all connections are closed and  in-flight requests cancelled
+        # Ensure all connections are closed and in-flight requests cancelled
         for conn in tuple(_dispatcher_map.values()):
             conn.close()
+        # The event loop should be closed, so we call remaining asyncore.close
+        # callbacks from here
+        self._timers.service_timeouts()
 
         log.debug("Dispatchers were closed")
```

Comment:
I believe this in an old bug, since if the `notify_loop` had been called just here, after all the dispatchers are closed and before completely exiting we would've have the same error